zenoh/src/net/routing/dispatcher/tables.rs

```diff
@@ -175,7 +175,9 @@ pub fn close_face(tables: &TablesLock, face: &Weak<FaceState>) {
             tracing::debug!("Close {}", face);
             face.task_controller.terminate_all(Duration::from_secs(10));
             finalize_pending_queries(tables, &mut face);
-            zlock!(tables.ctrl_lock).close_face(tables, &mut face);
+            let ctrl_lock = zlock!(tables.ctrl_lock);
+            finalize_pending_interests(tables, &mut face);
```

Comment:
Why do we need the CTRL lock finalize pending Interests but not Queries?

Reply:
In both cases, we need a lock to prevent concurrent insertion and removal of pending queries/interests. Queries have their dedicated lock to prevent that. I considered interest as control messages and so reused the control lock (already used at insertion anyway) for interests.